Source/src/BitStream.cpp

```diff
@@ -486,58 +486,74 @@ void BitStream::SetData( unsigned char *inByteArray )
 void BitStream::WriteCompressed( const unsigned char* inByteArray,
 								const unsigned int size, const bool unsignedData )
 {
-	BitSize_t currentByte = ( size >> 3 ) - 1; // PCs
+	BitSize_t currentByte;
 
-	unsigned char byteMatch;
-
-	if ( unsignedData )
-	{
-		byteMatch = 0;
-	}
-
-	else
-	{
-		byteMatch = 0xFF;
-	}
+	unsigned char byteMatch = unsignedData ? 0 : 0xFF;
 
 	// Write upper bytes with a single 1
-	// From high byte to low byte, if high byte is a byteMatch then write a 1 bit. Otherwise write a 0 bit and then write the remaining bytes
-	while ( currentByte > 0 )
+	// From high byte to low byte, if high byte is a byteMatch then write a 1 bit. 
+	// Otherwise write a 0 bit and then write the remaining bytes
+	if (!IsNetworkOrder)
 	{
-		if ( inByteArray[ currentByte ] == byteMatch )   // If high byte is byteMatch (0 of 0xff) then it would have the same value shifted
+		// get the highest byte with highest index  PCs
+		currentByte = (size >> 3) - 1;
+		while (currentByte > 0)
 		{
-			bool b = true;
-			Write( b );
+			if (inByteArray[currentByte] == byteMatch)   // If high byte is byteMatch (0 of 0xff) then it would have the same value shifted
+			{
+				Write(true);
+				currByte--；

```

Comment:
should be currentByte I take it
also invalid character used (0x203a) instead of ;